modules/storages/app/services/storages/group_folder_properties_sync_service.rb

```diff
@@ -212,7 +212,7 @@ def project_folder_permissions(project:)
       tokens_query = tokens_query.where(users: project.users)
     end
     tokens_query.each_with_object({
-                                    users: admins_project_folder_permissions,
+                                    users: admins_project_folder_permissions.clone,
```

Comment:
@ba1ash `admins_project_folder_permissions` is a memoized object. We need to clone it first, if we want to modify the object as we do in line 219. Othwerwise we also modify the memoized object and that would be a bad side effect.